nodeJS/orms/prisma_orm.md

```diff
@@ -0,0 +1,157 @@
+### Introduction
+
+Tired of writing raw SQL queries in your projects? In this lesson, we'll talk about Object Relational Mappers (ORM for short). ORMs are tools that let you manipulate data in your database and are widely used in the software industry. We'll dive deep into one ORM popular in the Node.js landscape: [Prisma ORM](https://www.prisma.io/).
+
+### Lesson overview
+
+This section contains a general overview of topics that you will learn in this lesson.
+
+- What are ORMs, and why are they used?
+- Introduction to Prisma ORM
+- Features of Prisma
+
+### Challenges with raw SQL'in
+
+With the couple of projects you've done earlier, you might've run into the tediousness of raw SQL queries. Let's briefly discuss some of these challenges.
+
+#### So much more code
+
+You need a select statement? Cool, write a query. Need a select statement for a different table? Write another query. Or if your programmer senses are tingling, you could create a utility function for select. But now you need a select statement for specific columns? You'll have to modify the utility function. What about filters and sorting? You get the point.
+
+Replicate the above for insert queries, and all of its variations. Or you could go in a different direction. Create modules per entity. For example, the books module would have:
+
+```javascript
+class Book {
+  async getBooks(filters) {}
+  async getBookById(id) {}
+  async createBook(data) {}
+  async updateBook(id, data) {}
+  async deleteBook(id) {}
+  async getBookAuthors(id) {}
+  async getBookGenres(id) {}
+  // and so on
+}
+```
+
+Or you could mix and match the two approaches. Create a `Database` class and inherit from it in your entity classes. Or you fancy composition, and just want to deal with pure functions.
+
+Now do this over and over for each of entity, across multiple projects. You get the point. It can be argued that the above is not a bad thing, as you're learning and practicing, and it isn't. It's sufficient for solo projects, but as your codebase grows, and as you work in teams, you'll realize how much an ORM really helps you focus on writing code that's business critical.
+
+Writing more code is not necessarily a bad thing, and we even suggest that you explore these above ideas in your solo projects. But as you start to work in teams, and with large scale software, it becomes imperative to have a standard way of interacting with the database, either with an external library or a homegrown solution.
+
+<div class="lesson-note lesson-note--tip" markdown="1">
+
+If you haven't applied any of the above paradigms in your previous projects, we highly recommend going back and refactoring. You might end up with a very basic mini ORM for your project. This will help you appreciate a bit more on how a proper ORM will make your life easier.
```

Comment:
Remove the `on` typo here:

> This will help you appreciate a bit more on how a proper ORM will make your life easier.

Should be:

> This will help you appreciate a bit more how a proper ORM will make your life easier.

Because the below doesn't read properly:

> This will help you appreciate ~~a bit more~~ on how a proper ORM will make your life easier.